src/main/webapp/app/exercises/programming/shared/code-editor/actions/code-editor-actions.component.ts

```diff
@@ -89,6 +93,14 @@ export class CodeEditorActionsComponent implements OnInit, OnDestroy, OnChanges
             .getBuildingState()
             .pipe(tap((isBuilding: boolean) => (this.isBuilding = isBuilding)))
             .subscribe();
+
+        this.autoSaveInterval = window.setInterval(() => {
+            this.autoSaveTimer++;
+            if (this.autoSaveTimer >= 30) {
+                this.autoSaveTimer = 0;
+                this.onSave();
+            }
+        }, 1000);
```

Comment:
It might be worth to have the autosave time stored in constants

Reply:
Done!